npm/qsharp/src/compiler/compiler.ts

```diff
@@ -273,10 +278,41 @@ export class Compiler implements ICompiler {
     );
   }
 
-  async getDocumentation(): Promise<IDocFile[]> {
+  // Returns all autogenerated documentation files for the standard library.
+  // This include file names and metadata, including specially formatted
+  // table of content file.
+  async getDocumentationFiles(): Promise<IDocFile[]> {
     return this.wasm.generate_docs();
   }
 
+  // Returns autogenerated documentation for the standard library
+  // and loaded project (if requested). Only the array of documentation
+  // content is returned.
+  async getDocumentationContent(
+    additionalSources?: Array<any>[],
+    targetProfile?: string,
+    languageFeatures?: string[],
+  ): Promise<string[]> {
+    // Get documentation from wasm layer
+    const docFiles: IDocFile[] = this.wasm.generate_docs(
+      additionalSources,
+      targetProfile,
+      languageFeatures,
+    );
+
+    const result: string[] = [];
+    for (const file of docFiles) {
```

Comment:
FWIW I think this is too much business logic for this layer. As much as possible, `compiler.ts` should call the wasm methods and pass the result through without modification, otherwise it becomes really hard to follow what's happening where.

I'm in favor of returning to having just one `getDocumentation()` method (with the new, optional parameters) , and moving this filtering to `vscode/src/documentation.ts`, the only current place it's used.

Keeping this layer simple will also minimize API breaking changes (though I don't think this is used by QCOM or anything today).

Reply:
I was thinking about unifying these two get functions but wanted to put some logic in there... Getting "files" from this API isn't convenient for anything other than just placing these files in the file system (which generation of the actual documentation does). I need to thing about this a bit more...

Reply:
I decided to keep single getDocumentation function like it used to be, only add three optional parameters to it.